### PR TITLE
Adr logging monitoring

### PR DESCRIPTION
**Pull request by vigneshmsft**
## Context

ADR for logging and monitoring

## Link to Trello card

https://trello.com/c/JcqcwjvE/2442-write-an-adr-on-our-approach-to-alerting-and-monitoring

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: adr/0014-alerting-monitoring.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+- Smoke tests [run after each deploy](https://github.com/DFE-Digital/apply-for-teacher-training-tests). These tests sign in to the service and make sure that the service is usable.
+- Sentry [tells us when exceptions occur](https://sentry.io/organizations/dfe-bat).  
+
+## Decision
+
+We've discussed expanding the things we alert for. For example, we could alert on lower-level technical issues like disk space, memory leaks, or high CPU. These alerts could potentially give us advance warning of infrastructure issues.
+
+However, they would also increase the possibility of false positives. <<MORE>>>
+
+The team is comfortable with the current availability monitoring setup in Azure and being alerted in the Slack channel in the event of an anomaly.
```
Comment (duncanjbrown) [inline adr/0014-alerting-monitoring.md:31]:
```suggestion

We've discussed expanding the things we alert for. For example, we could alert on lower-level technical issues like disk space, memory leaks, or high CPU. These alerts could potentially give us advance warning of infrastructure issues.

## Decision

**We will not add monitoring for low-level system metrics** as we have confidence in Axure (it's been running fine for months now) and we have confidence our current smoke tests and availability tests will pick up major problems. If we did find, for example, that memory leakage was at the root of an outage, we would add monitoring for that particular metric.

**We will continue to be alerted in Slack** for service availability and Sentry exceptions, and we will continue to improve our smoke tests
```

File: adr/0014-alerting-monitoring.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+- Smoke tests [run after each deploy](https://github.com/DFE-Digital/apply-for-teacher-training-tests). These tests sign in to the service and make sure that the service is usable.
+- Sentry [tells us when exceptions occur](https://sentry.io/organizations/dfe-bat).  
+
+We've discussed expanding the things we alert for. For example, we could alert on lower-level technical issues like disk space, memory leaks, or high CPU. These alerts could potentially give us advance warning of infrastructure issues.
+
+## Decision
+
+**We will not add monitoring for low-level system metrics** as we have confidence in Axure (it's been running fine for months now) and we have confidence our current smoke tests and availability tests will pick up major problems. If we did find, for example, that memory leakage was at the root of an outage, we would add monitoring for that particular metric.
```
Comment (vigneshmsft) [inline adr/0014-alerting-monitoring.md:29]:
```suggestion
**We will not add monitoring for low-level system metrics** as we have confidence in Azure (it's been running fine for months now) and we have confidence our current smoke tests and availability tests will pick up major problems. If we did find, for example, that memory leakage was at the root of an outage, we would add monitoring for that particular metric.
```

File: adr/0014-alerting-monitoring.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+## Status
+
+Accepted
+
+## Context
+
+The team have recently discussed Apply's approach to alerting. By alerting we mean that we are notified when:
+
+1. There's is an outage
+2. When the service is partially down, like DfE Signin or a specific page
```
Comment (vigneshmsft) [inline adr/0014-alerting-monitoring.md:14]:
```suggestion
2. The service is partially down, like DfE Signin or a specific feature
```

**Comment by duncanjbrown** [comment]
Rebased. I'm happy with this if you are @tijmenb @vigneshmsft ?

**Comment by vigneshmsft** [comment]
LGTM.

**Comment by duncanjbrown** [comment]
If you mark ready for review I'll approve @vigneshmsft

**Pull request merged**